src/main/groovy/util/Node.java

```diff
@@ -537,26 +539,40 @@ private NodeList getByName(String name) {
 
     /**
      * Provides a collection of all the nodes in the tree
-     * using a depth first traversal.
+     * using a depth-first preorder traversal.
      *
      * @return the list of (depth-first) ordered nodes
      */
     public List depthFirst() {
+        return depthFirst(true);
+    }
+
+    /**
+     * Provides a collection of all the nodes in the tree
+     * using a depth-first traversal.
+     *
+     * @param preorder if false, a postorder depth-first traversal will be performed
```

Comment:
I am not sure I understand where you are going with the BST idea but I have tweaked the tests a little as per your earlier comments.